sw/device/silicon_creator/lib/drivers/rstmgr_functest.c

```diff
@@ -0,0 +1,53 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#include <stdbool.h>
+#include <stdint.h>
+
+#include "sw/device/lib/runtime/log.h"
+#include "sw/device/lib/testing/check.h"
+#include "sw/device/silicon_creator/lib/drivers/retention_sram.h"
+#include "sw/device/silicon_creator/lib/drivers/rstmgr.h"
+#include "sw/device/silicon_creator/lib/error.h"
+#include "sw/device/silicon_creator/lib/test_main.h"
+
+const test_config_t kTestConfig;
+
+/**
+ * Test phases tracked in retention RAM.
+ */
+enum {
+  kTestPhaseInit = 0,
+  kTestPhaseReset = 1,
+  kTestPhaseDone = 2,
```

Comment:
Optional: we probably don't need all these test phases (one magic value that can be checked would probably suffice) but I know the other tests are written this way so might be best to leave it to keep everything consistent.

Reply:
Yes, initially had a single value but changed to follow the same pattern with existing tests for the sake of consistency.